### PR TITLE
Loop graph with set loop domain

### PR DESCRIPTION
**Pull request by naoyam**
Depends on #3036 and #3079
Part of #2851 

I'm still experimenting with IdModel setLoopDomain, but this PR at least
allows for correct loop graph and promotion maps to be generated as long
as no forwarding-like scheduling is involved in loop domains that are
not fully derived from logical domains.

The existing slice tests with setLoopDomain are extended with inlineMost
since it's now fully working.

Note that there is still an issue of inlining analysis (#3068).
Addressing it is not in the scope of this PR.

### CODE REVIEW

**Comment by naoyam** [comment]
!build

**Comment by zasdfgbnm** [comment]
@naoyam Could you resolve conflict? Looks like this PR contains diff from other PRs.

File: tests/cpp/test_resize.cpp [hunk 1]
```diff
@@ -3681,6 +3681,9 @@ TEST_F(ResizeTest, SliceScheduledLikeProducer) {
 
   std::vector<int64_t> shape({100});
 
+  EnableOptionsGuard enable_options_guard;
```
Comment (naoyam) [inline tests/cpp/test_resize.cpp:3684]: Just moved up as this option needs to be enabled before using `inlineMost`

File: tests/cpp/test_resize.cpp
Comment (naoyam) [inline tests/cpp/test_resize.cpp:1]: The only change in this file is effectively just adding `inlineMost`

**Comment by naoyam** [comment]
!build

**Comment by naoyam** [comment]
@zasdfgbnm This is ready for review

**Review by zasdfgbnm — APPROVED** [review]

**Pull request merged**